src/permissions.psm1

```diff
@@ -0,0 +1,448 @@
+
+    $local:AccessPolicyId = Resolve-SafeguardAccessPolicyId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure $PolicyToGet
+    $local:AccessPolicy = Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "AccessPolicies/$($local:AccessPolicyId)"
+    return $($local:AccessPolicy).AccessRequestProperties
+}
+
+<#
+.SYNOPSIS
+Get session properties of an access policies in Safeguard via the Web API.
+
+.DESCRIPTION
+Session properties are the settings confugured for sessions access request.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER PolicyToGet
+An integer containing the ID or a string containing the name of the access policy for which session properties to get.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Get-SafeguardAccessPolicySessionProperty testAccessPolicy
+
+.EXAMPLE
+Get-SafeguardAccessPolicySessionProperty 123
+#>
+function Get-SafeguardAccessPolicySessionProperty
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$PolicyToGet
+    )
+
+    $local:AccessPolicy
+    $ErrorActionPreference = "Stop"
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    $local:AccessPolicyId = Resolve-SafeguardAccessPolicyId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure $PolicyToGet
+    $local:AccessPolicy = Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "AccessPolicies/$($local:AccessPolicyId)"
+    return $($local:AccessPolicy).SessionProperties
+}
+
+<#
+.SYNOPSIS
+Get roles in Safeguard via the Web API.
+
+.DESCRIPTION
+Role or entitlement is a set of access request policies that restrict system access to authorized users
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER RoleToGet
+An integer containing the ID  or a string containing the name of the entitlement to get.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Get-SafeguardRole -AccessToken $token -Appliance 10.5.32.54 -Insecure
+
+.EXAMPLE
+Get-SafeguardRole testEntitlement
+
+.EXAMPLE
+Get-SafeguardRole 123
+#>
+function Get-SafeguardRole
```

Comment:
Change Role to Entitlement.  We want to use Entitlement everywhere instead of Role.  Eventually this will change in the API.